### PR TITLE
[civ2][doc/2] move install_ray into container parent class

### PR DESCRIPTION
**Pull request by can-anyscale**
Currently the tester_container install ray into itself before running test. Move this capability to the container parent class. doc also requires this ability

Test:
- CI

### CODE REVIEW

File: ci/ray_ci/container.py [hunk 1]
```diff
@@ -49,6 +49,26 @@ def run_script(self, script: List[str]) -> None:
...
+        env = os.environ.copy()
+        env["DOCKER_BUILDKIT"] = "1"
+        subprocess.check_call(
+            [
+                "docker",
+                "build",
+                "--build-arg",
+                f"BASE_IMAGE={self._get_docker_image()}",
+                "-t",
+                self._get_docker_image(),
```
Comment (aslonnie) [inline ci/ray_ci/container.py:62]: so.. we still have the `/rayci already exist` issue?

add a `--pull` maybe?

**Comment by can-anyscale** [comment]
Added `--pull`

File: .buildkite/core.rayci.yml [hunk 2]
```diff
@@ -8,12 +8,12 @@ steps:
         --workers "$${BUILDKITE_PARALLEL_JOB_COUNT}" --worker-id "$${BUILDKITE_PARALLEL_JOB}" --parallelism-per-worker 3
         --except-tags debug_tests,asan_tests,post_wheel_build,xcommit
     depends_on: corebuild
-    job_env: corebuild
+    job_env: forge
```
Comment (can-anyscale) [inline .buildkite/core.rayci.yml:11]: change these base to avoid 5 minutes of pulling corebuild

File: ci/ray_ci/tester.py [hunk 3]
```diff
@@ -71,11 +72,11 @@ def main(
     if not bazel_workspace_dir:
         raise Exception("Please use `bazelisk run //ci/ray_ci`")
     os.chdir(bazel_workspace_dir)
+    docker_login(_DOCKER_ECR_REPO.split("/")[0])
```
Comment (can-anyscale) [inline ci/ray_ci/tester.py:75]: this docker_login not requires since we pull the docker image from the internet; previously, it just uses the local cache

**Review by aslonnie — COMMENTED** [review]
will this make too many repeated execution (compiling of ray) when there is a code change?

**Comment by can-anyscale** [comment]
Ah good call, repeating compiling ray is properly not avoidable since it's the common denominator on all of these builds  and we have flipped it to the last step in civ2 to be able to cache other stuffs which take longer to run.

That said, we can properly cache even the ray build itself so that these repeating compilation are no-op.

Would be in another PR though, this issue already exists before this PR.

**Review by aslonnie — APPROVED** [review]

**Pull request merged**